### PR TITLE
docs: update phone auth

### PR DESCRIPTION
**Pull request by kangmingtay**
## What kind of change does this PR introduce?
* Add more details on how to update / verify a user's phone number 
* Remove description about the password being truncated because this issue has been fixed already (https://github.com/supabase/gotrue/pull/1368)
* Main gist of what was added:

### CODE REVIEW

**Comment by charislam** [comment]
Thanks Kang Ming! I looked at the whole phone login section and there's a lot of duplication across these pages (and it was a bit odd to add code samples on the main page when the details used to live in the provider pages)

Reorganized the section a bit, you might want to take a look to make sure it all still makes sense

File: apps/docs/pages/guides/auth/phone-login.mdx [hunk 1]
```diff
@@ -82,37 +239,61 @@ To verify the one-time password (OTP) sent to the user's phone number, call [`ve
...
+
+If successful the user will now be logged in and you should receive a valid session like:
+
+```json
+{
+  "access_token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJhdXRoZW50aWNhdGVkIiwiZXhwIjoxNjI3MjkxNTc3LCJzdWIiOiJmYTA2NTQ1Zi1kYmI1LTQxY2EtYjk1NC1kOGUyOTg4YzcxOTEiLCJlbWFpbCI6IiIsInBob25lIjoiNjU4NzUyMjAyOSIsImFwcF9tZXRhZGF0YSI6eyJwcm92aWRlciI6InBob25lIn0sInVzZXJfbWV0YWRhdGEiOnt9LCJyb2xlIjoiYXV0aGVudGljYXRlZCJ9.1BqRi0NbS_yr1f6hnr4q3s1ylMR3c1vkiJ4e_N55dhM",
+  "token_type": "bearer",
+  "expires_in": 3600,
+  "refresh_token": "LSp8LglPPvf0DxGMSj-vaQ"
+}
```
Comment (kangmingtay) [inline apps/docs/pages/guides/auth/phone-login.mdx:293]: hopefully these are mock values and don't actually contain information about a user / project

**Comment by kangmingtay** [comment]
@charislam the rest of the changes LGTM, i've helped to clean up the twilio / messagebird guides abit too - thanks for helping out!

File: apps/docs/pages/guides/auth/phone-login.mdx (hunk 1 above)
Comment (charislam) [reply apps/docs/pages/guides/auth/phone-login.mdx:293]: mock I assume, I copied them over from the old guides, but good point, I'll swap out for placeholders

**Review by charislam — APPROVED** [review]

**Pull request merged**